File: twilio-audio-router/src/main/java/com/twilio/audio_manager/AudioDevice.java
package com.twilio.audio_manager;

import androidx.annotation.NonNull;

public class AudioDevice {
    public final @NonNull String name;
    public final @NonNull Type type;

    public AudioDevice(@NonNull Type type, @NonNull String name) {
        this.type = type;
        this.name = name;
    }

    /** Audio device types */
    public enum Type {
        SPEAKERPHONE,
        WIRED_HEADSET,
        EARPIECE,
        BLUETOOTH
    }

    @Override
    public boolean equals(Object o) {
        if (o == this) {
            return true;
        } else if (!(o instanceof AudioDevice)) {
            return false;
        } else {
            return name.equals(((AudioDevice) o).name) && type.equals(((AudioDevice) o).type);
        }
    }
}

File: audioswitch/src/main/java/com/twilio/audioswitch/selection/AudioDevice.kt
package com.twilio.audioswitch.selection

/**
 * This class represents a single audio device that has been retrieved by the [AudioDeviceSelector].
 */
sealed class AudioDevice {

    /** The friendly name of the device.*/
    abstract val name: String

    /** An [AudioDevice] representing a Bluetooth Headset.*/
    data class BluetoothHeadset internal constructor(override val name: String) : AudioDevice()

    /** An [AudioDevice] representing a Wired Headset.*/
    data class WiredHeadset internal constructor(override val name: String = "Wired Headset") : AudioDevice()

    /** An [AudioDevice] representing the Earpiece.*/
    data class Earpiece internal constructor(override val name: String = "Earpiece") : AudioDevice()

    /** An [AudioDevice] representing the Speakerphone.*/
    data class Speakerphone internal constructor(override val name: String = "Speakerphone") : AudioDevice()
}
